frontend/src/journal/hooks/useJournal.ts

```diff
@@ -0,0 +1,191 @@
+import { useCallback, useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import {
+  useCreateJournalMutation,
+  useGetJournalQuery,
+  useGetJournalsQuery,
+  useRemoveJournalMutation,
+  useUpdateJournalMutation,
+} from "@/graphql/generated";
+
+const handleError = (error: unknown, message: string): void => {
+  console.error(message, error);
+};
+
+export function useCreateJournal() {
+  const { refetch } = useGetJournalsQuery();
+  const [createJournal] = useCreateJournalMutation();
+  const navigate = useNavigate();
+
+  const [journalId, setJournalId] = useState<string | null>(null);
+
+  const handleCreateNewJournal = async (): Promise<void> => {
+    try {
+      const response = await createJournal({
+        variables: { title: "Untitled", content: "What’s on my mind" },
+      });
+
+      const newJournalId = response.data?.createJournal?.id;
+      if (newJournalId) {
+        setJournalId(newJournalId);
+        refetch();
+        navigate(`/journals/${newJournalId}`);
+      }
+    } catch (error) {
+      handleError(error, "Error creating journal:");
+    }
+  };
+
+  return {
+    journalId,
+    handleCreateNewJournal,
+  };
+}
+
+export function useUpdateJournal() {
+  const { id } = useParams<{ id: string }>();
+  const { loading, error, data } = useGetJournalQuery({
+    variables: { id: id || "" },
+  });
+
+  const [updateJournal] = useUpdateJournalMutation();
+  const [title, setTitle] = useState<string>("");
+  const [content, setContent] = useState<string>("");
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const [debounceTimer, setDebounceTimer] = useState<NodeJS.Timeout | null>(
+    null
+  );
+
+  useEffect(() => {
+    if (data?.getJournal) {
+      setTitle(data.getJournal.title);
+      setContent(data.getJournal.content ?? "");
+    }
+  }, [data]);
+
+  const saveJournal = useCallback(
+    async (newTitle: string, newContent: string): Promise<void> => {
+      if (!id) return;
+      setIsSaving(true);
+      try {
+        await updateJournal({
+          variables: { id, title: newTitle, content: newContent },
+        });
+      } catch (err) {
+        console.error("Error saving journal:", err);
```

Comment:
Since you have `handleError` method — use it here too

Reply:
Done!